src/Helper/MyParcelCollection.php

```diff
@@ -625,6 +608,53 @@ public function clearConsignmentsCollection()
         $this->items = [];
     }
 
+    /**
+     * To search and filter consignments by certain values
+     *
+     * @param string $apiKey
+     * @param mixed  $parameters May be an array or object containing properties.
+     *                           If query_data is an array, it may be a simple one-dimensional structure,
+     *                           or an array of arrays (which in turn may contain other arrays).
+     *                           If query_data is an object, then only public properties will be incorporated
+     *                           into the result.
+     *
+     * @return \MyParcelNL\Sdk\src\Helper\MyParcelCollection
+     * @throws \MyParcelNL\Sdk\src\Exception\ApiException
+     * @throws \MyParcelNL\Sdk\src\Exception\MissingFieldException
+     * @throws \Exception
+     */
+    public static function query(string $apiKey, $parameters): MyParcelCollection
+    {
+        $collection = new static();
+
+        // The field `size` is required to prevent bugs. Think carefully about what
+        // the maximum size should be in your use case. If you want to pick up all
+        // open consignments for example, you would probably want to adjust size to 300.
+        if (empty($parameters['size'])) {
+            throw new MissingFieldException('Field size is requested.');
```

Comment:
Beetje vage zin, misschien `Field "size" is required.`?

Reply:
Done